src/lib/controllers/upload-controller.js

```diff
@@ -76,4 +77,27 @@ export default class UploadController {
       targetDocSources
     }
   }
+
+  static plainSourceUploadFromFileSingle ({ fileData, textId, textType, tokenization }) {
+    if (fileData.length === 0 || fileData.search(/\r\n|\r|\n/) === -1) {

```

Comment:
does this prevent the user from uploading a text with just a single line in it? I think we should allow that.

Reply:
Yes, it sets the requrement, that an uploaded file should contain:
1. text
2. lang
3. direction
4. textType

That's why it needs 4 lines at least

Reply:
The format is updated (details in the next comment)

Reply:
I see, I think we have different understandings of what the plain text upload should be.  I was thinking that the user would upload individual files with just the text they want to align, and use the user interface options to specify the details of lang, direction, segmentation etc.

They would upload an individual file for each text, not mix the origin and target texts(s) in the same file.

The only upload file format with multiple texts in it that I think we need to support for the first release is one which is already aligned (partially or fully) and which corresponds to the alpheios alignment schema (probably both the 1.0 version and a new 2.0 version which still needs to be fully defined).

They might choose to upload one text (whether origin and target) and use the text area to enter the text of another.

Reply:
oh, yes, I understood your wrong then.
ok - I will update `plainSourceUploadFromFileSingle` method with the ability to upload only text

Reply:
ok, thank you.